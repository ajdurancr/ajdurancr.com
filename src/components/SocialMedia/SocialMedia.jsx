import React from 'react'
import propTypes from 'prop-types'
import get from 'lodash.get'

import { Component } from '..'
import styles from './SocialMedia.module.sass'

const SocialMedia = ({ socialMediaLinks }) => (
  <div className={styles.container}>
    {socialMediaLinks.map((component) => (
      <Component key={get(component, 'sys.id')} {...component} />
    ))}
  </div>
)

SocialMedia.defaultProps = {
  socialMediaLinks: [],
}

SocialMedia.propTypes = {
  socialMediaLinks: propTypes.arrayOf(propTypes.object),
}

export default SocialMedia
