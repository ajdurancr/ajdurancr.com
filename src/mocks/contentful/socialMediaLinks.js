import media from './media'

export const socialMediaLink = {
  sys: {
    id: '7fb8b29Sx8xzEerHpUkFuC',
    contentType: {
      sys: {
        id: 'component',
      },
    },
  },
  fields: {
    name: 'Social Media Link - ajdurancr - GitHub',
    contentType: 'Social Media Link',
    props: [
      {
        sys: {
          id: '3DOnmJjojmOqGUCwZVXcmZ',
          contentType: {
            sys: {
              id: 'propMedia',
            },
          },
        },
        fields: {
          name: 'icon',
          value: {
            sys: {
              id: '5Yt9H0zn6ewSMnuvzh7V6P',
              locale: 'en-US',
            },
            fields: media,
          },
        },
      },
      {
        sys: {
          id: '4c1KdByPrvFRuKMILrEayJ',
          contentType: {
            sys: {
              id: 'propCta',
            },
          },
        },
        fields: {
          name: 'cta',
          text: 'GitHub',
          value: 'https://github.com/ajdurancr',
        },
      },
    ],
  },
}

export default [socialMediaLink]
