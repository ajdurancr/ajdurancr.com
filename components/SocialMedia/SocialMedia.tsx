import React from 'react'
import get from 'lodash.get'

import { Component } from '../Component'
import styles from './SocialMedia.module.sass'
import { CfComponent } from '../../types/contentful'

const SocialMedia: React.FC<{ socialMediaLinks: CfComponent[] }> = ({ socialMediaLinks }) => (
  <div className={styles.container}>
    {socialMediaLinks?.map((component) => (
      <Component key={get(component, 'sys.id')} component={component} />
    ))}
  </div>
)

export { SocialMedia }
