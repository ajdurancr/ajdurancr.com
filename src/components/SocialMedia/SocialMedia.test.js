import React from 'react'
import { shallow } from 'enzyme'

import SocialMedia from './SocialMedia'
import socialMediaLinks from '../../mocks/contentful/socialMediaLinks'

describe('SocialMedia', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SocialMedia socialMediaLinks={socialMediaLinks} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly, socialMediaLinks:falsy', () => {
    const wrapper = shallow(<SocialMedia />)

    expect(wrapper).toMatchSnapshot()
  })
})
