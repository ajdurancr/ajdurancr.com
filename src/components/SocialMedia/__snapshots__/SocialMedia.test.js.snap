// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia renders correctly 1`] = `
<div
  className="container"
>
  <Component
    fields={
      Object {
        "contentType": "Social Media Link",
        "name": "Social Media Link - ajdurancr - GitHub",
        "props": Array [
          Object {
            "fields": Object {
              "name": "icon",
              "value": Object {
                "fields": Object {
                  "description": "Github Logo",
                  "file": Object {
                    "contentType": "image/jpeg",
                    "details": Object {
                      "image": Object {
                        "height": 400,
                        "width": 400,
                      },
                      "size": 11536,
                    },
                    "fileName": "GitHub_filename",
                    "url": "//images.ctfassets.net/52uxn1pv2fou/5Yt9H0zn6ewSMnuvzh7V6P/3d6a253c1dce0250c70ef3a52c169872/github.jpg",
                  },
                  "title": "Media - GitHub Logo",
                },
                "sys": Object {
                  "id": "5Yt9H0zn6ewSMnuvzh7V6P",
                  "locale": "en-US",
                },
              },
            },
            "sys": Object {
              "contentType": Object {
                "sys": Object {
                  "id": "propMedia",
                },
              },
              "id": "3DOnmJjojmOqGUCwZVXcmZ",
            },
          },
          Object {
            "fields": Object {
              "name": "cta",
              "text": "GitHub",
              "value": "https://github.com/ajdurancr",
            },
            "sys": Object {
              "contentType": Object {
                "sys": Object {
                  "id": "propCta",
                },
              },
              "id": "4c1KdByPrvFRuKMILrEayJ",
            },
          },
        ],
      }
    }
    key="7fb8b29Sx8xzEerHpUkFuC"
    sys={
      Object {
        "contentType": Object {
          "sys": Object {
            "id": "component",
          },
        },
        "id": "7fb8b29Sx8xzEerHpUkFuC",
      }
    }
  />
</div>
`;

exports[`SocialMedia renders correctly, socialMediaLinks:falsy 1`] = `
<div
  className="container"
/>
`;
